topics/statistics/tutorials/intro-to-ml-with-python/tutorial.md

```diff
@@ -0,0 +1,4133 @@
+
+A simple **decision tree** reduces your problem into a **hierarchichal sequence of questions** on your features that can be answered by yes or no and which subdivides the data into 2 subgroups on which a new question is asked, and so on and so on.
+
+<!--![tree_ex](image/tree_ex.png)-->
+
+Ok, but a huge number of trees can actually be built just by considering the different orders of questions asked. How does the algorithm deals with this?
+
+Quite simply actually: it **tests all the features and chooses the most discriminative** (with respect to your target variable) : the feature where a yes or no question divides the data into 2 subsets which minimizes an **impurity measure**.
+
+Imagine you have a dataset with feature color (red or blue) and feature shape (square or circle), and 2 target classes : 1 and 2.
+
+
+<!--![tree](image/Tree.png)-->
+
+Asking `"feature color is red"` gives you the following subgroups:
+ * 10 class 1, and 1 class 2 (`"feature color is red" == True`)
+ * 2 class 1, and 11 class 2 (`"feature color is red" == False`)
+
+Asking `"feature shape is square"` gives you:
+ * 5 class 1, and 7 class 2 (`True`) 
+ * 7 class 1 and 5 class 2 (`False`)
+ 
+ So, you will prefer asking `"feature color is red?"` over `"feature shape is square?"`: `"feature color is red?"` is more discriminative.
+
+For **categorical variables, the questions test for a specific category**.
+For **numerical variables, the questions use a threshold** to as a yes/no question.  
+
+The **threshold is, again, chosen to minimize impurity**. And in turn the best threshold for a variable is used to estimate the discriminativeness of that variable.
+
+Of course, you will have to compute this threshold at each step of your tree since at each step you are considering different subdatasets.
+
+---
+The **impurity is related to how much your feature splitting is still having mixed classes**. So the impurity ends up giving a score: either it is a simple [Shannon entropy](https://en.wikipedia.org/wiki/Entropy_(information_theory)) or it is a [Gini coefficient](https://en.wikipedia.org/wiki/Gini_coefficient).
+
+#### Shannon Entropy
+
+$$Entropy = - \sum_{j} p_j log_2(p_j)$$
+
+This measure is linked to information theory, where the information of an event occuring is the $log_2$ of this event's probability of occuring.
+For purity, **0 is the best possible score, and 1 the worst**.
+
+#### Gini coefficient
+
+$$Gini = 1- \sum_{j} p_j^2$$
+
+The idea is to measure the **probability that a dummy classifier mislabels your data**.
+**0 is best, 1 is worst.**
+
+---
+Before going further, just a little bit of vocabulary: 
+* **Trees** are made of **nodes** (where the question is asked and where the splitting occurs). 
+* A **branch** is the outcome of a splitting. 
+* A **leaf** is the last node on a branch (no more splitting).
+
+[Back to the ToC](#toc)
+
+### Toy example to visualize decision tree. <a class="anchor" id="toy-decision-tree"></a>
```

Comment:
the build errors are due to manual anchors like this, just remove them and use the auto-generated anchors to refer to sections.